scripts/generatePackage.js

```diff
@@ -0,0 +1,70 @@
+import fsExtra from 'fs-extra';
+import fs from 'fs';
+import initPromptSync from 'prompt-sync';
+import { execSync } from 'child_process';
+import path from 'path';
+import { fileURLToPath } from 'url';
+import sortPackageJson from 'sort-package-json';
+import templatePackageJson from './package-template/package.json';
+import prettier from 'prettier';
+import chalk from 'chalk';
+
+// eslint-disable-next-line no-underscore-dangle
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+const prompt = initPromptSync();
+
+const templatePath = './scripts/package-template';
+const packagesFolder = './packages';
+
+(async () => {
+    const packageFolderName =
+        process.argv?.[2] || prompt(chalk.bold('Enter package name (without @trezor prefix): '));
+    if (!packageFolderName) process.exit(1);
+
+    const packageName = `@trezor/${packageFolderName}`;
+    const packagePath = `${packagesFolder}/${packageFolderName}`;
+
+    const workspaces = Object.keys(
+        JSON.parse(JSON.parse(execSync('yarn workspaces --json info').toString()).data),
+    );
+
+    if (fs.existsSync(packagePath)) {
+        console.error(
+            chalk.bold.red(`Folder ${packagePath} already exists! Please choose different name.`),
+        );
+        process.exit(1);
+    }
+
+    if (workspaces.includes(packageName)) {
+        console.error(
+            chalk.bold.red(`Package ${packageName} already exists! Please choose different name.`),
+        );
+        process.exit(1);
+    }
+
+    const packageJson = {
+        ...templatePackageJson,
+        name: packageName,
+    };
+
+    const prettierConfigPath = await prettier.resolveConfigFile();
+    const prettierConfig = {
+        ...(await prettier.resolveConfig(prettierConfigPath)),
+        parser: 'json',
+    };
+
+    const serializeConfig = config =>
+        prettier.format(JSON.stringify(config).replace(/\\\\/g, '/'), prettierConfig);
+
+    try {
+        fsExtra.copySync(templatePath, packagePath);
```

Comment:
Why make it synchronous?
For this kind of tasks I think it is better to follow the asynchronous nature of nodejs.

Do you think having it sync brings some benefit?

Reply:
It's much easier to work with, no reason to complicate life here with async things, also I need to wait to complete this task anyway before doing anything else. It takes max few ms to complete so we really don't have to care about perf here.

Reply:
I understand that using synchronous blocking methods seams to allow a faster development but it usually leads to a very weird and confusing issues, so I usually avoid it as a good practice.

Specially you can use some util like [promisify](https://nodejs.org/api/util.html#utilpromisifyoriginal) that allow you to use it with async/await improving the readability.

That being said, if you really like that, I guess it is a small script and shouldn't be a big issue.